forem/forem
# RFC#50-P2 Restructuring the sidebar to add nested nav items + tabbed nav items

- [x]  I need to fix content_manager section: organizations, podcasts and tags
- [x]  need to refactor
- [x]  fix code climate issues 
- [x]  update QA section on this PR + screenshot fo navbar

## What type of PR is this? (check all applicable)


## Description
Please note that this is Part Two of a multiple series PR. I prefer to make the incremental substantial changes to the code to receive feedback earlier and  to test the changes in Production under the feature flag rather than develop the whole feature in one PR. 

In the [previous PR](https://github.com/forem/forem/pull/12851/files) I had shuffled the routes around to prepare the routing for admin navigation restructure.

In this PR, I've done the following: 
- Created a new data structure to handle the nested routes (up to the third level)
- Displayed the route structure in the sidebar in a nested manner and linked them to the correct routes
- Created a partial that gets brought into the view when there is a third level of navigation


In this PR, I've _not_ yet tackled the following:
- Add the feature flagged routes (like profile and DUS routes) to the Navigation bar 
- Add the icons 
- Make the nav items expand and collapse
- Added the Interaction limits route
- Highlight the current route in the sidebar

## Related Tickets & Documents
https://github.com/forem/rfcs/pull/50

## QA Instructions, Screenshots, Recordings

<img width="1680" alt="Screenshot 2021-03-04 at 16 49 30" src="https://user-images.githubusercontent.com/2786819/109982683-cf89ed80-7d0a-11eb-9e05-5356c7765101.png">

- Firstly and most importantly ensure that none of the new admin routes and the nested navigation does not show up when the feature flag `:admin_restructure` is disabled.
- Now activate the feature flag through http://localhost:3000/admin/feature_flags/features and you should see the updated nav elements
- Navigate through the sidebar and click links around, ensure that the navigation items have second and third level items as per [the doc](https://github.com/forem/rfcs/blob/aefc141f386b395cc12d4b9f56816f1de4702eab/text/0050-update-admin-ia-nav.md#3-move-existing-content-into-new-location) and the [mock](https://www.figma.com/file/auEpfZTbdeXdjN5OIXxQ2K/Admin-IA-and-Nav?node-id=219%3A5605).

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
- [x] This change does not need to be communicated, and this is why not: will be communicated at the end

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/l7fdqmHQ1jCg2HzQlx/giphy.gif)
